lte/gateway/c/session_manager/SessionStateEnforcer.cpp

```diff
@@ -632,47 +633,56 @@ void SessionStateEnforcer::prepare_response_to_access(
-      config.rat_specific_context.m5gsm_session_context()
-          .default_ambr()
-          .max_bandwidth_ul());
-  rsp->mutable_session_ambr()->set_max_bandwidth_dl(
-      config.rat_specific_context.m5gsm_session_context()
-          .default_ambr()
-          .max_bandwidth_dl());
-  /* This flag is used for sending defult qos value or getting from policy
-   *  value to AMF.
+  /* If there are any static rules configured from Orc8r via NMS it's
+   * corresponding qos info will be send to AMF.
    */
-  bool flag_set = false;
   for (auto& val : pending_activation) {
-    if (val.rule.qos().max_req_bw_dl() && val.rule.qos().max_req_bw_ul()) {
+    if (val.rule.has_qos()) {
       MLOG(MDEBUG) << "value set for pending_activation"
                    << val.rule.qos().max_req_bw_ul();
-      rsp->mutable_session_ambr()->set_max_bandwidth_dl(
-          val.rule.qos().max_req_bw_dl());
-      rsp->mutable_session_ambr()->set_max_bandwidth_ul(
-          val.rule.qos().max_req_bw_ul());
       rsp->mutable_qos()->CopyFrom(val.rule.qos());
-      flag_set = true;
-      break;
     }
   }
-  if (!flag_set) {
-    auto* convg_qos = rsp->mutable_qos();
-    convg_qos->set_qci(FlowQos_Qci_QCI_9);
-    convg_qos->mutable_arp()->set_pre_vulnerability(
-        QosArp_PreVul_PRE_VUL_ENABLED);
-    convg_qos->mutable_arp()->set_pre_capability(QosArp_PreCap_PRE_CAP_ENABLED);
-    convg_qos->mutable_arp()->set_priority_level(1);
+  /* AMBR value need to compared from AMF and PCF, then fill the required
+   * values and sent to AMF. Otherwise sessiond will set default AMBR info.
+   */
+  if (config.rat_specific_context.m5gsm_session_context()
+          .has_subscribed_qos()) {
+    rsp->mutable_subscribed_qos()->set_br_unit(
+        config.rat_specific_context.m5gsm_session_context()
+            .subscribed_qos()
+            .br_unit());
+    rsp->mutable_subscribed_qos()->set_apn_ambr_ul(
+        config.rat_specific_context.m5gsm_session_context()
+            .subscribed_qos()
+            .apn_ambr_ul());
+    rsp->mutable_subscribed_qos()->set_apn_ambr_dl(
+        config.rat_specific_context.m5gsm_session_context()
+            .subscribed_qos()
+            .apn_ambr_dl());
+    rsp->mutable_subscribed_qos()->set_priority_level(
+        config.rat_specific_context.m5gsm_session_context()
+            .subscribed_qos()
+            .priority_level());
+    rsp->mutable_subscribed_qos()->set_preemption_capability(
+        config.rat_specific_context.m5gsm_session_context()
+            .subscribed_qos()
+            .preemption_capability());
+    rsp->mutable_subscribed_qos()->set_preemption_vulnerability(
+        config.rat_specific_context.m5gsm_session_context()
+            .subscribed_qos()
+            .preemption_vulnerability());
+    rsp->mutable_subscribed_qos()->set_qos_class_id(
+        config.rat_specific_context.m5gsm_session_context()
+            .subscribed_qos()
+            .qos_class_id());
+

```

Comment:
this should be pulled into a separate function to keep this one easy to read

Reply:
Moved to separate function as suggested.